src/realm/object-store/list.cpp

```diff
@@ -379,7 +379,19 @@ bool List::operator==(List const& rgt) const noexcept
 
 List List::freeze(std::shared_ptr<Realm> const& frozen_realm) const
 {
-    return List(frozen_realm, frozen_realm->import_copy_of(*m_coll_base));
+    return List(frozen_realm, *frozen_realm->import_copy_of(*m_coll_base));
```

Comment:
It is possible for `frozen_realm->import_copy_of(*m_coll_base)` to return null here if the source Realm is in a write transaction and this List is newly created. Currently this results in it returning a `List` where `is_valid()` is false, and this changes it to instead be UB. In the valid case it's adding a pointless extra copy of the collection object.

Reply:
Yes, this is very unfortunate conceptual poo poo. I'd like to be able to "freeze" in the middle of a write transaction to eliminate it, but that is a much larger effort.

Reply:
Fixed.